src/jailer/src/env.rs

```diff
@@ -130,6 +130,14 @@ impl Env {
         let exec_file_name = exec_file_path
             .file_name()
             .ok_or_else(|| Error::FileName(exec_file_path.clone()))?;
+        let exec_file_name_as_string = exec_file_name
+            .to_str()
+            // Safe to unwrap as the original `exec_file` is `String`.
+            .unwrap()
+            .to_string();
+        if !exec_file_name_as_string.contains("firecracker") {
+            return Err(Error::ExecFileName(exec_file_name_as_string));
+        }
```

Comment:
Nit: I would put all checks around this (from line 123 to 140) to a separate function, to improve readability.

Reply:
If this is only done here it might be easiest to use a scope.
```rust
// Validates file
{
    // your checks here
    // ...
}
```

Reply:
Even better. Let's use scope!

Reply:
OK. I'll revise it using scope.

Reply:
Sorry, @zulinx86, your comment wasn't yet visible when I commented :sweat_smile:

Reply:
No worries, @luminitavoicu !
Currently, I'm thinking of coding as follows:
```
@@ -130,14 +130,7 @@ impl Env {
         let exec_file_name = exec_file_path
             .file_name()
             .ok_or_else(|| Error::FileName(exec_file_path.clone()))?;
-        let exec_file_name_as_string = exec_file_name
-            .to_str()
-            // Safe to unwrap as the original `exec_file` is `String`.
-            .unwrap()
-            .to_string();
-        if !exec_file_name_as_string.contains("firecracker") {
-            return Err(Error::ExecFileName(exec_file_name_as_string));
-        }
+        Env::validate_exec_file(exec_file_name)?;
 
         let chroot_base = arguments
             .single_value("chroot-base-dir")
@@ -255,6 +248,18 @@ impl Env {
         self.uid
     }
 
+    fn validate_exec_file(exec_file_name: &OsStr) -> Result<()> {
+        let exec_file_name_as_string = exec_file_name
+            .to_str()
+            // Safe to unwrap as the original `exec_file` is `String`.
+            .unwrap()
+            .to_string();
+        match exec_file_name_as_string.contains("firecracker") {
+            true => Ok(()),
+            false => Err(Error::ExecFileName(exec_file_name_as_string)),
+        }
+    }
+
     fn parse_resource_limits(resource_limits: &mut ResourceLimits, args: &[String]) -> Result<()> {
         for arg in args {
             let (name, value) = arg
@@ -887,6 +892,15 @@ mod tests {
         }
     }
 
+    #[test]
+    fn test_validate_exec_file() {
+        // Success case.
+        assert!(Env::validate_exec_file(OsStr::new("firecracker_xyz")).is_ok());
+
+        // Error case: filename without "firecracker".
+        assert!(Env::validate_exec_file(OsStr::new("foobarbaz")).is_err());
+    }
+
     #[test]
     fn test_setup_jailed_folder() {
         let mut mock_cgroups = MockCgroupFs::new().unwrap();
```
Splitting function looks easier to test.
How do you think about this?

Reply:
After discussing with @luminitavoicu , I decided to adopt splitting it into a function.
- Pros of using scope.
  - It's simple.
  - We can avoid adding a new function which leads to jump to it when reading.
- Pros of splitting it into a function.
  - It's easier to test the functionality that I added this time.
  - We can keep `Env::new()` small for readability.
  - We can include the canonicalization part of `exec_file` into the function, which results in comprehensive tests.

Thanks for advice!